Split Action Types and AIConfigReducer into it's own folder

rossdanlm opened the pull request:
Split Action Types and AIConfigReducer into it's own folder

I just feel having the action type definitions vs. the reducer implementation make this a lot easier to read. You can now have file side by side instead of having to jump back and forth across the same one. I also merged the run commands together under the `RunPromptAction` type to make things cleaner as well. This will make it easier next diff once I merge logic together for the run prompts

I also cleaned up the `UPDATE_PROMPT_INPUT` implementation inside of the aiConfigReducer because it wasn't passing in the correct input type anyways, so now that's cleaner too.

## Test Plan
Everything works the same as before

https://github.com/lastmile-ai/aiconfig/assets/151060367/46c11941-c055-4307-99b2-3cd6202b78cb

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/926).
* #928
* __->__ #926
* #925
* #924
* #922
* #921
* #920
* #919
* #918
* #917

rholinshead reviewed (approved):
> I also cleaned up the `UPDATE_PROMPT_INPUT` implementation inside of the aiConfigReducer because it wasn't passing in the correct input type anyways, so now that's cleaner too.

Can you explain what you mean? As far as I can tell the implementation here is the same except inline instead of a helper function, which is fine - we can add it back if we have other actions that update input later.

rossdanlm commented:
> > I also cleaned up the `UPDATE_PROMPT_INPUT` implementation inside of the aiConfigReducer because it wasn't passing in the correct input type anyways, so now that's cleaner too.
> 
> Can you explain what you mean? As far as I can tell the implementation here is the same except inline instead of a helper function, which is fine - we can add it back if we have other actions that update input later.

So the method `reduceReplacePrompt()` expects a replace function callback of type `(prompt) -> ClientPrompt`. Before this replace function was of type `() -> ClientPrompt`

<img width="1150" alt="Screenshot 2024-01-15 at 15 16 03" src="https://github.com/lastmile-ai/aiconfig/assets/151060367/4eb6d81c-33b2-4fe0-bd32-7685f3829ee4">

For some reason this error was only flagged to me once I moved around logic in this diff. Let me know if I'm missing anything

rholinshead commented:
> > > I also cleaned up the `UPDATE_PROMPT_INPUT` implementation inside of the aiConfigReducer because it wasn't passing in the correct input type anyways, so now that's cleaner too.
> > 
> > 
> > Can you explain what you mean? As far as I can tell the implementation here is the same except inline instead of a helper function, which is fine - we can add it back if we have other actions that update input later.
> 
> So the method `reduceReplacePrompt()` expects a replace function callback of type `(prompt) -> ClientPrompt`. Before this replace function was of type `() -> ClientPrompt`
> 
> <img alt="Screenshot 2024-01-15 at 15 16 03" width="1150" src="https://private-user-images.githubusercontent.com/151060367/296862371-4eb6d81c-33b2-4fe0-bd32-7685f3829ee4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzNTA3MDYsIm5iZiI6MTcwNTM1MDQwNiwicGF0aCI6Ii8xNTEwNjAzNjcvMjk2ODYyMzcxLTRlYjZkODFjLTMzYjItNGZlMC1iZDMyLTc2ODVmMzgyOWVlNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNVQyMDI2NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MWNiNTI2YjQ4MTQwZGY2MGU3OWIyZWNkYjEyMjYyYTkxMzIxYTMxY2EzY2JjNjc1YmJiNDQ1ZTIwNzVlZjM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Wl0b1HDEHrSibNdQpIPS_ANi1QcxfCTNRjKQfogJX3Q">
> For some reason this error was only flagged to me once I moved around logic in this diff. Let me know if I'm missing anything

In `reduceReplaceInput` we define that `(prompt) => ClientPrompt` function inline as:
```
(prompt) => ({
    ...prompt,
    input: replacerFn(prompt.input),
  })
```
so it's not an error

rossdanlm commented:
Ohhhh, thanks that makes sense! I think I just removed it to keep it cleaner. It felt a bit weird to me we were using this as a helper function when we already had the more generic `reduceReplacePrompt()`!
